ttnn/cpp/ttnn/operations/eltwise/unary/unary.cpp

```diff
@@ -305,6 +305,39 @@ Tensor Identity::invoke(
         DefaultQueueId, input_tensor, {UnaryWithParam{op_type}}, memory_config, optional_output_tensor);
 }
 
+Tensor Dropout::invoke(
+    const Tensor& input,
+    const uint32_t seed,
+    const float probability,
+    const float scale,
+    const std::optional<MemoryConfig>& memory_config,
+    const std::optional<Tensor>& optional_output_tensor) {
+    TT_ASSERT(input.device()->arch() != tt::ARCH::GRAYSKULL, "Dropout is not currently supported on Grayskull");
+    return detail::unary_impl(
+        DefaultQueueId,
+        input,
+        {UnaryWithParam{UnaryOpType::DROPOUT, {static_cast<float>(seed), probability, scale}}},
+        memory_config,
+        optional_output_tensor);
+}
+
+Tensor Dropout::invoke(
+    uint8_t queue_id,
+    const Tensor& input,
+    const uint32_t seed,
+    const float probability,
+    const float scale,
+    const std::optional<MemoryConfig>& memory_config,
+    const std::optional<Tensor>& optional_output_tensor) {
+    TT_ASSERT(input.device()->arch() != tt::ARCH::GRAYSKULL, "Dropout is not currently supported on Grayskull");
+    return detail::unary_impl(
+        queue_id,
+        input,
+        {UnaryWithParam{UnaryOpType::DROPOUT, {static_cast<float>(seed), probability, scale}}},
```

Comment:
Same question as others, why is the seed in float ?

Reply:
It is the limitation of the current unary op implementation. They are using vector<float> to pass parameters.
I cannot do much here.

Reply:
In my opinion all these ops should be rewritten.  But it is out of scope of the current pr.

Reply:
I do agree with you here.  Unfortunately we have a freeze on the API at the moment unless this blocks models or customers.